tests/test_gce.py

```diff
@@ -1,27 +1,44 @@
+from pytest import fixture
+from unittest.mock import MagicMock, patch
 import json
 from datetime import datetime, timezone, timedelta
-from ocw.lib.gce import GCE, Provider
+from ocw.lib.gce import GCE
 from webui.PCWConfig import PCWConfig
 from tests.generators import max_age_hours, mock_get_feature_property
 
 
-class FakeRequest:
+class MockRequest:
     def __init__(self, response={}):
         self.response = response
 
     def execute(self):
         return self.response
 
 
-class FakeMockRegions:
+class MockResource:
     def __init__(self, responses):
+        self.deleted_resources = list()
         self.responses = responses
 
+    def __call__(self, *args, **kwargs):
+        return self
+
     def list(self, *args, **kwargs):
-        return self.responses.pop(0)
+        if len(self.responses) > 0:
+            return self.responses.pop(0)
```

Comment:
we don't need this, but I understand if we have >0 could be more than one, so pop all of them?